cmd/podman/patch.go

```diff
@@ -0,0 +1,203 @@
+    }
+
+    if ! canPatch() {
+        fmt.Println("Patch command not on your system")
+        os.Exit(1)
+    }
+
+    if ! exists(patch) {
+        fmt.Println("Patch file not found:", patch)
+        os.Exit(1)
+    }
+
+    return patchContainers(runtime, patch, dst, containers, c.Ignore)
+}
+
+// Retrieve all running containers and return array of name
+func retrieveExisting(runtime *libpod.Runtime) ([]string) {
+    runningPods, err := runtime.GetRunningContainers()
+    if err != nil {
+        fmt.Println(err)
+        os.Exit(1)
+    }
+    var containers []string
+
+    for _, container := range runningPods {
+        containers = append(containers, container.Name())
+    }
+    return containers
+}
+
+// Check if the patch command is available on the current system (host)
+func canPatch() (bool) {
+    _, err := exec.Command("which", "patch").Output()
+    if err != nil {
+        return false
+    }
+    return true
+}
+
+// Check if a file exist
+func exists(path string) (bool) {
+    _, err := os.Stat(path)
+    if os.IsNotExist(err) {
+        return false
+    }
+    if err == nil { return true }
+    return true
+}
+
+// Retrieve all the containers by name and
+// loop over to apply the given patch
+func patchContainers(runtime *libpod.Runtime,
+                     patch string, dst string, containers []string, ignore bool) error {
+    var ctrs []*libpod.Container
+    for _, container := range containers {
+        ctr, err := runtime.LookupContainer(container)
+        if err != nil {
+            fmt.Println(err)
+            os.Exit(1)
+        }
+        ctrs = append(ctrs, ctr)
+    }
+    for _, ctr := range ctrs {
+        applyPatch(patch, dst, ctr, ignore)
+    }
+    return nil
+}
+
+// Apply the given patch file to the given container.
+// The patch is applied to the dst file.
+// If the destintion doesn't exist inside the given container
+// and if the flag ignore-fail is given the command
+// simply continue and doesn't exit with an error code.
+func applyPatch(patch string, dst string, container *libpod.Container, ignore bool) {
+    fmt.Println("Patching: ", container.Name())
+    mountPoint, err := container.Mount()
+    if err != nil {
+        fmt.Println(err)
+        os.Exit(1)
+    }
+
+    fulldest := filepath.Join(mountPoint, dst)
+
+    if ! exists(fulldest) {
+        fmt.Println("File to patch not found:", dst)
+        if ignore {
+            fmt.Println("Skipping container:", container.Name())
+            return
+        } else {
+            os.Exit(1)
+        }
+    }
+    fmt.Println("Execute: patch ", fulldest, patch)
+    cmd := exec.Command("patch", fulldest, patch)
+    var out bytes.Buffer
+    var stderr bytes.Buffer
+    cmd.Stdout = &out
+    cmd.Stderr = &stderr
+    errexec := cmd.Run()
+    if errexec != nil {
+        fmt.Println(fmt.Sprint(errexec) + ": " + stderr.String())
+    }
+    container.Unmount(true)
+    fmt.Println(container.Name(), "patched successfully!")
```

Comment:
I think we probably just want to print out the container ID here, similar to how Start and other commands work - error printed on error, full container ID on success